operators/vision/image_encoder_win32.hpp

```diff
@@ -0,0 +1,225 @@
+    if (FAILED(hr)) {
+      return errorWithHr_("Failed to create pIWICFactory.", hr);
+    }
+
+    return {};
+  }
+
+  bool JpgSupportsBgr() const{ return true; }
+
+  void EncodeJpg(const uint8_t* source_data, bool source_is_bgr, int32_t width, int32_t height,
+                uint8_t** outbuffer, size_t* outsize) const {
+    std::vector<PROPBAG2> options;
+    std::vector<VARIANT> values;
+
+    {
+      PROPBAG2 option = {0};
+      option.pstrName = L"ImageQuality";
+      options.push_back(option);
+      VARIANT varValue;
+      VariantInit(&varValue);
+      varValue.vt = VT_R4;
+      varValue.fltVal = static_cast<FLOAT>(0.95);
+      values.push_back(varValue);
+    }
+
+    {
+      PROPBAG2 option = {0};
+      option.pstrName = L"JpegYCrCbSubsampling";
+      options.push_back(option);
+      VARIANT varValue;
+      VariantInit(&varValue);
+      varValue.vt = VT_UI1;
+      varValue.bVal = WICJpegYCrCbSubsampling420;
+      values.push_back(varValue);
+    }
+
+    return EncodeWith(GUID_ContainerFormatJpeg, options, values, source_data, source_is_bgr,
+        width, height, outbuffer,outsize);
+  }
+
+  bool pngSupportsBgr() const{ return true; }
+
+  void EncodePng(const uint8_t* source_data, bool source_is_bgr, int32_t width, int32_t height,
+                uint8_t** outbuffer,size_t* outsize) const{
+    std::vector<PROPBAG2> options;
+    std::vector<VARIANT> values;
+
+    {
+      PROPBAG2 option = {0};
+      option.pstrName = L"InterlaceOption";
+      options.push_back(option);
+      VARIANT varValue;
+      VariantInit(&varValue);
+      varValue.vt = VT_BOOL;
+      varValue.boolVal = FALSE;
+      values.push_back(varValue);
+    }
+
+    {
+      PROPBAG2 option = {0};
+      option.pstrName = L"FilterOption";
+      options.push_back(option);
+      VARIANT varValue;
+      VariantInit(&varValue);
+      varValue.vt = VT_UI1;
+      varValue.bVal = WICPngFilterSub;
+      values.push_back(varValue);
+    }
+
+    return EncodeWith(GUID_ContainerFormatPng, options, values, source_data, source_is_bgr, width, height,
+                      outbuffer, outsize);
+  }
+
+  ~EncodeImage() {
+    pIWICFactory_.detach();
+    CoUninitialize();
+  }
+
+  private:
+   void EncodeWith(GUID conatinerFormatGUID, std::vector<PROPBAG2> options, std::vector<VARIANT> values,
+                   const uint8_t* source_data, bool source_is_bgr, int32_t width, int32_t height, uint8_t** outbuffer,
+                   size_t* outsize) const {
+     const uint8_t* encode_source_data = source_data;
+     winrt::com_ptr<IStream> pOutputStream;
+     winrt::com_ptr<IWICBitmapEncoder> pIEncoder;
+     winrt::com_ptr<IWICBitmapFrameEncode> pBitmapFrame;
+     winrt::com_ptr<IPropertyBag2> pPropertyBag;
+
+     HRESULT hr = ::CreateStreamOnHGlobal(NULL, FALSE, pOutputStream.put());
+     if (FAILED(hr)) {
+       ORTX_CXX_API_THROW(errorWithHr_("Failed when CreateStreamOnHGlobal.", hr).Message(), ORT_RUNTIME_EXCEPTION);
```

Comment:
In the existing `libjpeg` and `libpng` encoder, `ORTX_CXX_API_THROW` was used. So I copied this behaviour. Should this be a `OrtxStatus` instead (like in decoders)? Should this use `ORT_RUNTIME_EXCEPTION`? I'm not sure

Reply:
How can we get if the format is png or jpeg, without using `BaseKernel`? Is it still possible?